docs/src/pages/guides/composition/composition.md

```diff
@@ -96,3 +99,17 @@ class ListItemLink extends React.Component {
 Here is a demo with [React Router](https://github.com/ReactTraining/react-router):
 
 {{"demo": "pages/guides/composition/ComponentProperty.js"}}
+
+### Caveat with refs
+Some components such as `ButtonBase` (and therefore `Button`) require access to the 
+underlying DOM node. This was previously done with `ReactDOM.findDOMNode(this)`.
+However `findDOMNode` was deprecated (which disqualifies its usage in react's concurrent mode)
+in favour of component refs and ref forwarding. If you pass class components to
+the `component` and don't run in strict mode you won't have to change anything
+since we can safely use `ReactDOM.findDOMNode`. For function components you have
+to wrap your component however in `React.forwardRef`:
+
+```diff
+- <Button component={props => <div {...props} />} />
++ <Button component={React.forwardRef((props, ref) => <div {...props} ref={ref} />)} />
```

Comment:
Better show the component creation outside of the render method.

Reply:
This is just a concise overview for the change. It's not real world usage. I don't think extending the diff over 2-3 lines is actual helpful but I will check how it looks.

Reply:
I was thinking of something like this:

```diff
-const MyButton = props => <div {...props} />;
+const MyButton = React.forwardRef((props, ref) => <div {...props} ref={ref} />);

<Button component={MyButton} />
```

Reply:
That's way better 👍